### ISSUE TITLE
[DOC]: Add some clarification about ZeroInitContext, ColoInitContext, and colossalai.initialize

### ISSUE THREAD

**Issue by JThh** [hpcaitech/ColossalAI#2840]
### 📚 The doc issue

There is a issue raise in the slack channel about their differences. I have posted a [preliminary answer](https://colossalaiworkspace.slack.com/archives/C02NAJARJ9Y/p1676959657443859?thread_ts=1676900155.658179&cid=C02NAJARJ9Y) to it based on my inspection of our source codes. Please polish it and make it accessible in our official documentation system. Thanks!

**Comment by FrankLeeeee** [comment]
Great suggestion. In fact, we are trying to refactor this repository a bit in the future. We will release our design and development plan soon.

**Comment by laozhanghahaha** [comment]
> Great suggestion. In fact, we are trying to refactor this repository a bit in the future. We will release our design and development plan soon.

Hey I'm also confused about the difference between coloinitcontext and zeroinitcontext. I noticeed in the example/ dir, you use different initcontext for different language model, could you clarify the differences?

And I can not open the link https://colossalaiworkspace.slack.com/archives/C02NAJARJ9Y/p1676959657443859?thread_ts=1676900155.658179&cid=C02NAJARJ9Y

**Comment by JThh** [comment]
FYI,

Hi, with ColoInitContext and with ZeroInitContext  are not compatible and you may use either of them. ColoInitContext  initializes the torch module with coloparameters that contain distributed specifications. ZeroInitContext  auto-shards the parameters onto devices according to your provided strategy.

colossalai.initialize is another orthogonal method that wraps torch model and dataloader around to enable users to use our own training loops. It is mainly for ease of use.

To summarize, you may try out ZeroInitContext when using ZeRO strategies (ddp) and ColoInitContext  when making tensor parallelism. You can always use colossalai.initialize  for ease of training.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by laozhanghahaha** [comment]
@JThh  Thanks for your explaination, and I still have some questions
1. `ZeroInitContext  auto-shards the parameters onto devices according to your provided strategy` If I use GeminiDDP (which support Zero and gemini) combined with coloinitcontext, will it shard parameters?

2. I try to load HF model with ColoinitContext and it works, but when I switch to ZeroInitContext, It ends with cuda oom.  It is a little bit weired to me, since I thought zeroinitcontext could shard parameters and use less GPU memory per device. Why would this happen?
